provider/cloudflare/cloudflare.go

```diff
@@ -806,9 +846,10 @@ func groupByNameAndTypeWithCustomHostnames(records DNSRecordsMap, chs CustomHost
 			continue
 		}
 		ep = ep.WithProviderSpecific(source.CloudflareProxiedKey, strconv.FormatBool(proxied))
-		// noop (customOriginServers is empty) if custom hostnames feature is not in use
-		if customHostname, ok := customOriginServers[records[0].Name]; ok {
-			ep = ep.WithProviderSpecific(source.CloudflareCustomHostnameKey, customHostname)
+		// noop (customHostnames is empty) if custom hostnames feature is not in use
+		if customHostnames, ok := customHostnames[records[0].Name]; ok {
+			sort.Strings(customHostnames)
```

Comment:
do we need sort so many times? We could use map or something, this will preserve order as example

Reply:
we only need to sort it twice:
1. when grabbing the custom hostnames from the annotations with expectedly random order (above, L581);
2. when scanning DNS records and associated custom hostnames in cloudflare and as a result assembling provider specific property in CSV format (for the quoted fragment here, L852);

i've removed the extra sort in the `getEndpointCustomHostnames()` itself since it doesn't matter and should be already sorted as a result of sort that is already in place

Reply:
ok